lib/jekyll-gist/gist_tag.rb

```diff
@@ -4,7 +4,12 @@ class GistTag < Liquid::Tag
 
       def render(context)
         if tag_contents = determine_arguments(@markup.strip)
-          gist_id, filename = tag_contents[0], tag_contents[1]
+          gist_id = context.registers[:page]['gist_id']
```

Comment:
I was thinking of this, I'll make a local gist_id have more priority over front matter, if there's a `gist_id` in front matter and another gist_id in the line it self, i should take the one in the line instead of the global one, if none provided then i take the global one.